### DISCUSSION TITLE
请问非齐次poisson过程有什么比较好的仿真方法吗

### DISCUSSION THREAD

**Discussion by Lodfy-Feng** [physics-data/faq#148] [Q&A] [answered]
尝试过生成齐次Possion过程后按照lambda(t)/lambda和random(0-1)进行比较后进行取舍，但效果奇差，如果非齐次Poisson同样具有时间独立性，即从t时刻计时起后仍然为Poisson过程，此算法似乎可行，但这一说法我也没有得以求证。 cry

**Comment by heroxbd** [comment, marked as answer]
有三种方案，前两种我用过，第三种没有试验过

1. 把 Poisson 强度对时间积分，将到这一时间窗口内的总计数期望，然后 Poisson 分布采样得到事例总数 N 。把时间强度流归一化，看成是每个事例的时间分布，独立赋予每个事例时间  t 。
2. 把 Poisson 强度在时间上分成若干个小区间，按每个小区间上的事例期望数独立进行 Poisson 分布采样，时间取为每个区间的中心。
3. 搜索 sampling from non-homogeneous Poisson process 的论文，预计可以得到比以上两种更一般更有效的模拟方法。

**Reply by Lodfy-Feng** [reply]
感谢续老师，我试试

**Comment by yuchenguommm** [comment]
太奇怪了，我就是按照你的方法模拟的，效果还不错啊。是不是你的模拟过程出现了什么问题，建议仔细及检查一下代码。
